docs/LRM-feedback.md

```diff
@@ -0,0 +1,420 @@
+
+## 2: Lexical Conventions 
+
+#### 2.1 Strings
+
+Strings are sequences of characters surrounded by double quotes `"`.
+As string has the type `str`.
+
+#### 2.2 Identifiers
+
+Identifiers are a sequence of characters consisting of uppercase or lowercase letters, digits, underscores 
+or dashes. 
+
+#### 2.3 Keywords
+
+The following identifiers are restricted from use:
+
+* `int`
+* `str`
+* `regx`
+* `if`
+* `else`
+* `elif`
+* `else`
+* `for`
+* `while`
+* `break`
+* `return`
+* `func`
+* `array`
+* `file`
+* `print`
+* `return`
+* `map`
+* `filter`
+* `true`
+* `false`
+* `null`
+* `void`
+
+
+#### 2.5 Regular Expressions
+
+Regular expressions are a special sequence of characters used for pattern matching surounded by single quotes `'` and preceded by the keyword `r'`. Regular expressions are of type `regx`.   
+
+#### 2.6 Comments
+
+FIRE only supports the use the block comments. Comments are initiated with the `/*` symbol and terminated by the `*/` symbol. Everything in between the symbols will be ignored by FIRE during compilation
+
+## 3: Meaning of Identifiers
+
+Identifiers are names that correlate to a single value, a function, or an array. The restrictions on valid identifiers are found in section 2.2.
+
+The scope of an identifier can be either global or local. A local identifier's scope is limited to inside the braces `{ ... }` in which they are declared, whereas global identifiers are declared outside any braces.
+
+## 4: Expressions
+
+#### 4.1 Primary Expressions
+
+Primary expressions in FIRE can either because accessing an array type or a function call. The primary expression is followed by an expression of either asignments in the case of a function call or an accessing of the datatypes in an array.
+
+1. `Primary Expression [ expression ]`
+2. `Primary Expression ( expression )`
+
+
+#### 4.2 Assignment Operator
+
+The assignment operators `=` returns the value of the expression that is evaluated on its right-hand side and stores it in the identifier on the left hand side. The scope of that identifier is described in [section 3](#Identifiers)
+
+#### 4.3 Function Calls
+
+Functions take in arguments by value except in the case of other functions which are passed by reference. Functions need to be assigned before being called but can be declared anonymously. An anonymous function's scope is within the function that it is declared.
+
+#### 4.4 Logical Negation
+
+TODO We're considering adding booleans, its easier to parse.
+
+Although FIRE does not directly support a boolean type, it equates `true` expressions to `1` and false expressions to `0`. As such, the logical negation operator `!` will convert the result of the logical expression on which it is applied to the inverse of what would normally be expected.
+
+All non-zero integers are evaluated as `true` where as `0` is false.
```

Comment:
You should make it explicit how you're actually implementing. If your boolean constants are just an alias for non-zero and zero integers, then you should say so. You should also make it clear that your conditional statements (`if` and `while`) only take `int` types. Something like `if ("fubar, brother") {...}` is not well-defined, and it probably should be.

Otherwise, you should strictly enforce that integers can't be used in place of booleans. (i.e. something like `if (1) { ... }` should not type check)

Reply:
Supporting booleans